### PR TITLE
Hooks for balances, orders, book and market

### PR DESCRIPTION
**Pull request by da-kami**
You had a good point there earlier @thomaseizinger 😁 - pretty sweet composing those hooks

### CODE REVIEW

**Review by thomaseizinger — APPROVED** [review]

**Review by D4nte — COMMENTED** [review]
Looks functionally the same 👍

**Comment by da-kami** [comment]
With the last 4 commits we add proper loading states. Since I did not want to tear down everything I added the loading state to the domain objects (i.e. `CurrencyValue`, `Market`).
We could opt for keeping this state outside these objects, but for now it was easier to deal with the functionality this way.
Tested the behavior in the UI and it is pretty smooth, due to caching it sometimes flickers between minus value and Spinner, but that is acceptable I'd say.

File: app/hooks/useBitcoinBalance.tsx [hunk 1]
```diff
@@ -1,20 +1,23 @@
...
-      errorRetryInterval: 500,
-      errorRetryCount: 10,
-      initialData: BigNumber.from(0)
+      refreshInterval: 10000
     }
   );
 
+  if (!balance || error) {
+    return {
+      ...ZERO_BTC,
```
Comment (D4nte) [inline app/hooks/useBitcoinBalance.tsx:17]: Could we not display any figure instead?

Comment (da-kami) [reply app/hooks/useBitcoinBalance.tsx:17]: We display a [Spinner](https://chakra-ui.com/spinner) if `isLoading` is set, no matter what the value is. I set the value to zero.

**Pull request merged**